pkg/asset-manager-utils/contracts/RebalancingRelayer.sol

```diff
@@ -34,8 +34,13 @@ contract RebalancingRelayer is IBasePoolRelayer, AssetHelpers {
         0x0000000000000000000000000000000000000000000000000000000000000001
     );
 
-    modifier rebalance(bytes32 poolId, IAsset[] memory assets) {
+    modifier rebalance(
+        bytes32 poolId,
+        IAsset[] memory assets,
+        uint256[] memory minCashBalances
+    ) {
         _require(_calledPool == _EMPTY_CALLED_POOL, Errors.REBALANCING_RELAYER_REENTERED);
+        _ensureCashBalance(poolId, _translateToIERC20(assets), minCashBalances);
```

Comment:
Let's store `tokens = _translate(assets)` as a variable, to avoid looping over this array twice.